programs/mango-v4/src/instructions/account_buyback_fees_with_mngo.rs

```diff
@@ -38,14 +38,16 @@ pub fn account_buyback_fees_with_mngo(
     let now_ts = clock.unix_timestamp.try_into().unwrap();
     let slot = clock.slot;
 
+    let mngo_oracle_ref = &AccountInfoRef::borrow(&ctx.accounts.mngo_oracle.as_ref())?;
     let mngo_oracle_price = mngo_bank.oracle_price(
-        &AccountInfoRef::borrow(&ctx.accounts.mngo_oracle.as_ref())?,
+        &OracleAccountInfos::from_reader(mngo_oracle_ref),
```

Comment:
This looks a bit annoying to me, because of the awkwardness of having to make an AccountInfoRef (which just stores a collection of references) and then store a _reference_ to that in OracleAccountInfos (now a collection of references to things that have references).

I suspect that `&AccountInfoRef` should also implement KeyedAccountReader and then a lot of issues would go away, because then we could probably make a `OracleAccountInfos::from_ai()` helper that returns a `OracleAccountInfos<&AccountInfoRef>` without needing to make a named temporary solely for ownership reasons.

I'd be up for making that change, doesn't need to be part of this PR.

Reply:
I played with it a bit and don't think there's a straightforward way to make this nicer. The key problem always is that AccountInfos have a RefCell and someone needs to hold on to the Ref instance if we want to have long-lived references to account data.